Automattic/jetpack
# Tiled Gallery Block: allow in Offline Mode.

#### Changes proposed in this Pull Request:

Classic Tiled Galleries (created via the Classic editor or within a Classic block in the block editor) can be used even when the site is in Offline mode, ever since #521.

The Tiled Gallery block, however, behaves differently; it requires an active connection. Let's try to be consistent with the feature by:

- Registering the Tiled Gallery block even when the site is in offline mode.
- Adding a new utility in the block editor to detect if a site is in offline mode.
- Short-circuiting and avoid trying to use Photon for Tiled gallery images in the block editor when a site is in offline mode.

#### Jetpack product discussion

* None, but this PR is a good follow-up from #13471 where we made the block not rely on the Photon module being active, and being more dynamic instead. 
* For reference, tiled gallery settings remain a bit confusing. #7576 is where this is discussed.

#### Does this pull request change what data or activity we track or use?

* No

#### Testing instructions:

I would suggest testing on two different sites:

* One that's connected to WordPress.com.
    * The Tiled Gallery block should remain available in the block editor.
    * When publishing a post with a tiled gallery, you should see it use Photon for its images once published.
* One that's using [Offline mode](https://jetpack.com/support/development-mode/)
    * The Tiled Gallery block should now be available in the block editor.
    * Galleries should look good in the editor and on the frontend.
    * Gallery images should not be using Photon.
